modules/renter/hostdb/hostentry_test.go

```diff
@@ -0,0 +1,110 @@
+package hostdb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// TestInsertHost tests the insertHost method.
+func TestInsertHost(t *testing.T) {
+	// no dependencies necessary
+	hdb := newHostDB(nil, nil, nil, nil, nil, nil)
+
+	// invalid host should not be scanned
+	hdb.insertHost(modules.HostSettings{NetAddress: "foo"})
+	select {
+	case <-hdb.scanPool:
+		t.Error("invalid host was added to scan pool")
+	case <-time.After(10 * time.Millisecond):
+	}
+
+	// valid host should be scanned
+	hdb.insertHost(modules.HostSettings{NetAddress: "foo:1234"})
+	select {
+	case <-hdb.scanPool:
+	case <-time.After(10 * time.Millisecond):
+		t.Error("host was not scanned")
+	}
+
+	// duplicate host should not be scanned
+	hdb.allHosts["bar:1234"] = nil
+	hdb.insertHost(modules.HostSettings{NetAddress: "bar:1234"})
+	select {
+	case <-hdb.scanPool:
+		t.Error("duplicate host was added to scan pool")
+	case <-time.After(10 * time.Millisecond):
+	}
+}
+
+// TestActiveHosts tests the ActiveHosts method.
+func TestActiveHosts(t *testing.T) {
+	// no dependencies necessary
+	hdb := newHostDB(nil, nil, nil, nil, nil, nil)
+
+	// empty
+	if hosts := hdb.ActiveHosts(); len(hosts) != 0 {
+		t.Errorf("wrong number of hosts: expected %v, got %v", 0, len(hosts))
+	}
+
+	// with one host
+	h1 := new(hostEntry)
+	h1.NetAddress = "foo"
+	hdb.activeHosts = map[modules.NetAddress]*hostNode{
+		h1.NetAddress: &hostNode{hostEntry: h1},
+	}
+	if hosts := hdb.ActiveHosts(); len(hosts) != 1 {
+		t.Errorf("wrong number of hosts: expected %v, got %v", 1, len(hosts))
+	}
+
+	// with multiple hosts
+	h2 := new(hostEntry)
+	h2.NetAddress = "bar"
+	hdb.activeHosts = map[modules.NetAddress]*hostNode{
+		h1.NetAddress: &hostNode{hostEntry: h1},
+		h2.NetAddress: &hostNode{hostEntry: h2},
+	}
+	if hosts := hdb.ActiveHosts(); len(hosts) != 2 {
+		t.Errorf("wrong number of hosts: expected %v, got %v", 2, len(hosts))
+	}
+}
```

Comment:
It should probably check that the contents of `hosts` match the contents of `hdb.activeHosts`. I agree that just checking the length isn't sufficient.